uiowa/uiowa
# Static map as media entity

Resolves https://github.com/uiowa/uiowa/issues/5367
<details>
<summary>Previous development details</summary>

Quick summary of where I'm leaving this at: 
- I haven't exported config, so when you pull this branch you'll need to go to https://housing.uiowa.ddev.site/admin/structure/media and create the static map media type (make sure to set Media Source as Static Map).
- Media Library and rendering is not working yet, you can test just adding a static map link on https://housing.uiowa.ddev.site/admin/content/media
- The 2 constraint/validation files will be needed but for now are commented out as they aren't working correctly
- the `docroot/modules/custom/sitenow_media_wysiwyg/css/media-overrides.css` and `docroot/modules/custom/sitenow_media_wysiwyg/sitenow_media_wysiwyg.libraries.yml` are most likely wrong, as nothing was rendering at all I delayed figuring out the right way to format them
- most of the static map files are identical to our panopto files, as I was trying to get just the basics working to start. `docroot/modules/custom/sitenow_media_wysiwyg/src/Plugin/Field/FieldFormatter/StaticMapUrlFormatter.php` and `docroot/modules/custom/sitenow_media_wysiwyg/src/Plugin/media/Source/StaticMap.php` borrow from other media types, though, and are probably why the media library doesn't work. When I change them to resemble panopto, though, I start getting `Plugin not found` errors.
</details>



# How to test


1. Pull down, composer install, all that jazz
2. Sync default site
3. create new page, use any wysiwyg and add a new media item
4. choose static map
5. paste the share link of any building on the uiowa map in the link field. (ex: https://maps.uiowa.edu/?id=1890#!m/579256)
6. fill out the other fields, (Tip: 17 is a good base zoom level)
7. verify thumbnail is created
8. Verify that a map is created
9. Verify that hovering over it shows a tooltip with the alt text
10. Verify that the aria-label of the link says that it will take you to maps.uiowa.edu
11. Verify that the `img` tag has alt text.
12. Edit media within WYSIWYG allows changing between Original (Medium Square), widescreen and ultrawide.
13. Test bad URLs (not maps.uiowa.edu base url, no id, no marker, incorrect marker). With dblog enabled check recent messages for an unable to create thumbnail message.
